File: Samples/Verbum/how-to-implement-the-insert-document-field-dialog-winforms-version-t223036-14.2.6-/VB/WindowsFormsApplication1/InsertFieldModules/ucPAGENUMBER.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Namespace WindowsFormsApplication1.InsertFieldModules
	Partial Public Class ucPAGENUMBER
		Inherits UserControl
		Implements IFieldParametersDialog
		Public Sub New(ByVal currentFieldName As String)
			InitializeComponent()
			If currentFieldName = "PAGE" Then
				InitializeListBoxItemsPage()
				labelControl1.Text = "At present the PAGE (NUMPAGES) field attributes are not supported." & Constants.vbCrLf
				labelControl1.Text &= "<href=https://www.devexpress.com/Support/Center/Question/Details/T221870>https://www.devexpress.com/Support/Center/Question/Details/T221870</href>"
				AddHandler labelControl1.HyperlinkClick, AddressOf labelControl1_HyperlinkClick
			Else
				InitializeListBoxItemsNumPages()
				layoutControlItem2.Visibility = DevExpress.XtraLayout.Utils.LayoutVisibility.Never
			End If

		End Sub

		Private Sub labelControl1_HyperlinkClick(ByVal sender As Object, ByVal e As DevExpress.Utils.HyperlinkClickEventArgs)
			System.Diagnostics.Process.Start(e.Link)
		End Sub

		Private Sub InitializeListBoxItemsPage()
			Dim listSource As New List(Of FieldAttributeItem)()
			listSource.Add(New FieldAttributeItem() With {.DisplayName = "1, 2, 3, ...", .AttributeValue = "\* Arabic"})
			listSource.Add(New FieldAttributeItem() With {.DisplayName = "- 1 -, - 2- , - 3- , ...", .AttributeValue = "\* ArabicDash"})
			listSource.Add(New FieldAttributeItem() With {.DisplayName = "a, b, c, ...", .AttributeValue = "\* alphabetic"})
			listSource.Add(New FieldAttributeItem() With {.DisplayName = "A, B, C, ...", .AttributeValue = "\* ALPHABETIC"})
			listSource.Add(New FieldAttributeItem() With {.DisplayName = "i, ii, iii, ...", .AttributeValue = "\* roman"})
			listSource.Add(New FieldAttributeItem() With {.DisplayName = "I, II, III, ...", .AttributeValue = "\* ROMAN"})
			listSource.Add(New FieldAttributeItem() With {.DisplayName = "Use Default Numbering", .AttributeValue = "EMPTY"})

			listBoxControlAttributes.DataSource = listSource
			listBoxControlAttributes.DisplayMember = "DisplayName"
			listBoxControlAttributes.ValueMember = "AttributeValue"
		End Sub

		Private Sub InitializeListBoxItemsNumPages()
			Dim listSource As New List(Of FieldAttributeItem)()
			listSource.Add(New FieldAttributeItem() With {.DisplayName = "(none)", .AttributeValue = "EMPTY"})
			listSource.Add(New FieldAttributeItem() With {.DisplayName = "1, 2, 3, ...", .AttributeValue = "\* Arabic"})
			listSource.Add(New FieldAttributeItem() With {.DisplayName = "a, b, c, ...", .AttributeValue = "\* alphabetic"})
			listSource.Add(New FieldAttributeItem() With {.DisplayName = "A, B, C, ...", .AttributeValue = "\* ALPHABETIC"})
			listSource.Add(New FieldAttributeItem() With {.DisplayName = "i, ii, iii, ...", .AttributeValue = "\* roman"})
			listSource.Add(New FieldAttributeItem() With {.DisplayName = "I, II, III, ...", .AttributeValue = "\* ROMAN"})
			listSource.Add(New FieldAttributeItem() With {.DisplayName = "1st, 2nd, 3rd, ...", .AttributeValue = "\* Ordinal"})
			listSource.Add(New FieldAttributeItem() With {.DisplayName = "One, Two, Three, ...", .AttributeValue = "\* CardText"})
			listSource.Add(New FieldAttributeItem() With {.DisplayName = "First, Second, Third, ...", .AttributeValue = "\* OrdText"})

			listBoxControlAttributes.DataSource = listSource
			listBoxControlAttributes.DisplayMember = "DisplayName"
			listBoxControlAttributes.ValueMember = "AttributeValue"
		End Sub


		#Region "IFieldParametersDialog Members"

		Public Function GenerateFieldParametersString() As String Implements IFieldParametersDialog.GenerateFieldParametersString
			Dim resultString As String = ""
			If listBoxControlAttributes.SelectedValue IsNot Nothing AndAlso listBoxControlAttributes.SelectedValue.ToString() <> "EMPTY" Then
				resultString = listBoxControlAttributes.SelectedValue.ToString()
			End If
			Return resultString
		End Function

		#End Region
	End Class
End Namespace

File: Samples/Verbum/how-to-implement-the-insert-document-field-dialog-winforms-version-t223036-14.2.6-/CS/WindowsFormsApplication1/InsertFieldModules/ucPAGENUMBER.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace WindowsFormsApplication1.InsertFieldModules {
    public partial class ucPAGENUMBER : UserControl, IFieldParametersDialog {
        public ucPAGENUMBER(string currentFieldName) {
            InitializeComponent();
            if(currentFieldName == "PAGE") {
                InitializeListBoxItemsPage();
                labelControl1.Text = "At present the PAGE (NUMPAGES) field attributes are not supported.\r\n";
                labelControl1.Text += "<href=https://www.devexpress.com/Support/Center/Question/Details/T221870>https://www.devexpress.com/Support/Center/Question/Details/T221870</href>";
                labelControl1.HyperlinkClick += labelControl1_HyperlinkClick;
            }
            else {
                InitializeListBoxItemsNumPages();
                layoutControlItem2.Visibility = DevExpress.XtraLayout.Utils.LayoutVisibility.Never;
            }
            
        }

        void labelControl1_HyperlinkClick(object sender, DevExpress.Utils.HyperlinkClickEventArgs e) {
            System.Diagnostics.Process.Start(e.Link);
        }

        void InitializeListBoxItemsPage() {
            List<FieldAttributeItem> listSource = new List<FieldAttributeItem>();
            listSource.Add(new FieldAttributeItem() { DisplayName = "1, 2, 3, ...", AttributeValue = @"\* Arabic"});
            listSource.Add(new FieldAttributeItem() { DisplayName = "- 1 -, - 2- , - 3- , ...", AttributeValue = @"\* ArabicDash" });
            listSource.Add(new FieldAttributeItem() { DisplayName = "a, b, c, ...", AttributeValue = @"\* alphabetic" });
            listSource.Add(new FieldAttributeItem() { DisplayName = "A, B, C, ...", AttributeValue = @"\* ALPHABETIC" });
            listSource.Add(new FieldAttributeItem() { DisplayName = "i, ii, iii, ...", AttributeValue = @"\* roman" });
            listSource.Add(new FieldAttributeItem() { DisplayName = "I, II, III, ...", AttributeValue = @"\* ROMAN" });
            listSource.Add(new FieldAttributeItem() { DisplayName = "Use Default Numbering", AttributeValue = @"EMPTY" });

            listBoxControlAttributes.DataSource = listSource;
            listBoxControlAttributes.DisplayMember = "DisplayName";
            listBoxControlAttributes.ValueMember = "AttributeValue";        
        }

        void InitializeListBoxItemsNumPages() {
            List<FieldAttributeItem> listSource = new List<FieldAttributeItem>();
            listSource.Add(new FieldAttributeItem() { DisplayName = "(none)", AttributeValue = @"EMPTY" });
            listSource.Add(new FieldAttributeItem() { DisplayName = "1, 2, 3, ...", AttributeValue = @"\* Arabic" });
            listSource.Add(new FieldAttributeItem() { DisplayName = "a, b, c, ...", AttributeValue = @"\* alphabetic" });
            listSource.Add(new FieldAttributeItem() { DisplayName = "A, B, C, ...", AttributeValue = @"\* ALPHABETIC" });
            listSource.Add(new FieldAttributeItem() { DisplayName = "i, ii, iii, ...", AttributeValue = @"\* roman" });
            listSource.Add(new FieldAttributeItem() { DisplayName = "I, II, III, ...", AttributeValue = @"\* ROMAN" });
            listSource.Add(new FieldAttributeItem() { DisplayName = "1st, 2nd, 3rd, ...", AttributeValue = @"\* Ordinal" });
            listSource.Add(new FieldAttributeItem() { DisplayName = "One, Two, Three, ...", AttributeValue = @"\* CardText" });
            listSource.Add(new FieldAttributeItem() { DisplayName = "First, Second, Third, ...", AttributeValue = @"\* OrdText" });

            listBoxControlAttributes.DataSource = listSource;
            listBoxControlAttributes.DisplayMember = "DisplayName";
            listBoxControlAttributes.ValueMember = "AttributeValue";
        }


        #region IFieldParametersDialog Members

        public string GenerateFieldParametersString() {
            string resultString = "";
            if(listBoxControlAttributes.SelectedValue != null && listBoxControlAttributes.SelectedValue.ToString() != "EMPTY") {
                resultString = listBoxControlAttributes.SelectedValue.ToString();
            }
            return resultString;            
        }

        #endregion
    }
}
